pkg/web/http.go

```diff
@@ -0,0 +1,69 @@
+package web
+
+import (
+	"bytes"
+	"crypto/tls"
+	"github.com/gorilla/websocket"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"time"
+)
+
+type WrappedSocketListener struct {
+	sockets chan *WrappedSocket
+	done    chan struct{}
+	addr    net.Addr
+}
+
+func (wsl *WrappedSocketListener) Accept() (net.Conn, error) {
+	select {
+	case ws := <-wsl.sockets:
+		return ws, nil
+	case <-wsl.done:
+		return nil, io.ErrClosedPipe
+	}
+}
+
+func (wsl *WrappedSocketListener) Close() error {
+	wsl.done <- struct{}{}
+	return nil
+}
+
+func (wsl *WrappedSocketListener) Addr() net.Addr {
+	return wsl.addr
+}
+
+var upgrader = websocket.Upgrader{
+	HandshakeTimeout: 20 * time.Second,
+	Subprotocols:     []string{"mumble1.2.5", "binary"},
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+}
+
+func NewHTTPServer(addr net.Addr, tlscfg *tls.Config, logger *log.Logger) (*http.Server, net.Listener) {
```

Comment:
Returning an `*http.Server` seems wrong here; it should be the caller's responsibility to create and manage the server.  I think it makes more sense to return a concrete type that implements an `http.Handler` and `net.Listener`.